maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/AbstractSurefireMojo.java

```diff
@@ -2292,23 +2291,52 @@ private void checkMethodFilterInIncludesExcludes( Iterable<String> patterns )
                 if ( pattern != null && pattern.contains( "#" ) )
                 {
                     throw new MojoFailureException( "Method filter prohibited in "
-                                                        + "includes|excludes|includesFile|excludesFile parameter: "
+                                                        + "includes|excludes parameter: "
                                                         + pattern );
                 }
             }
         }
     }
 
-    private TestListResolver getIncludedAndExcludedTests()
+    private TestListResolver solveIncludedAndExcludedTests()
```

Comment:
Method returns two values .... 

One by standard `return` statement and second by call `setTest(...)` which changes `Mojo` state and can cause impact to other place ... 

I didn't analyze flow where `solveIncludedAndExcludedTests` and `getTest` are used and what order are.

Try to describe what does this method do ...

Reply:
Updated in https://github.com/apache/maven-surefire/pull/400/commits/bc8768bccda866ee3edbb7c12a8f439ff351bcdd